src/azure-cli/azure/cli/command_modules/servicefabric/tests/latest/test_sf_commands.py

```diff
@@ -0,0 +1,95 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+import unittest
+import json
+import logging
+import os
+import time
+from azure.cli.core.util import CLIError
+from azure.cli.testsdk import ScenarioTest, LiveScenarioTest, ResourceGroupPreparer
+
+TEST_DIR = os.path.abspath(os.path.join(os.path.abspath(__file__), '..'))
+
+logger = logging.getLogger(__name__)
+
+
+def _create_keyvault(test, kwargs):
+    kwargs.update({'policy_path': os.path.join(TEST_DIR, 'policy.json')})
+
+    test.cmd('keyvault create --resource-group {rg} -n {kv_name} -l {loc} --enabled-for-deployment true --enabled-for-template-deployment true')
+    test.cmd('keyvault certificate create --vault-name {kv_name} -n {cert_name} -p @"{policy_path}"')
+
+
+def _create_cluster(test, kwargs):
+    _create_keyvault(test, kwargs)
+
+    while True:
+        cert = test.cmd('keyvault certificate show --vault-name {kv_name} -n {cert_name}').get_output_in_json()
+        if cert:
+            break
+    assert cert['sid'] is not None
+    cert_secret_id = cert['sid']
+    logger.error(cert_secret_id)
+    kwargs.update({'cert_secret_id': cert_secret_id})
+
+    test.cmd('az sf cluster create -g {rg} -n {cluster_name} -l {loc} --secret-identifier {cert_secret_id} --vm-password "{vm_password}" --cluster-size 3')
+    timeout = time.time() + 900
+    while True:
+        cluster = test.cmd('az sf cluster show -g {rg} -n {cluster_name}').get_output_in_json()
+
+        if cluster['provisioningState']:
+            if cluster['provisioningState'] == 'Succeeded':
+                return
+            if cluster['provisioningState'] == 'Failed':
+                raise CLIError("Cluster deployment was not succesful")
```

Comment:
Can we use check() here like https://github.com/Azure/azure-cli/blob/dev/doc/authoring_tests.md#sample-4-prepare-a-resource-group-for-a-test?

Reply:
I'm doing this way because it takes some time for provisioningState to get to Succeeded after the command finishes executing

Reply:
But it is show command, why does it take time to get provisioningState? Is it a long running operation?

Reply:
what takes a long time is the create. I'm using the show to check when the states changes to succeeded